### DISCUSSION TITLE
How do you connect a Trainer / HRM over Bluetooth?

### DISCUSSION THREAD

**Discussion by robd003** [dvmarinoff/Auuki#171] [Q&A]
New user here, I've tried click on pretty much everything in the Settings UI but I'm not getting prompted to scan for Bluetooth devices at all. I'm on the production demo page using Chromium on macOS.

What am I missing?

**Comment by dvmarinoff** [comment]
Hi, can you get me a screenshot of the app with the console open like that:
<img width="1438" alt="Screenshot 2023-12-16 at 12 01 19" src="https://github.com/dvmarinoff/Flux/assets/7523832/e5997b42-6900-4070-98a3-bdb40d6dbe58">

**Reply by dvmarinoff** [reply]
The console opens with right click -> inspect -> console tab

**Comment by robd003** [comment]
Turns out it's an issue with Brave globally banning web bluetooth.

Error:
```
device.js:76 :ble 'Could not request ble:controllable' DOMException: Web Bluetooth API globally disabled.
```

No problem with Flux, just need to get Brave to not be so paranoid...

**Reply by dvmarinoff** [reply]
So Brave is disabling web-bluetooth for all other browser? Didn't know that's possible. Will have to add it to the README and handle it in the UI too. Do you at least get the Red banner: `this browser in NOT supported ...`?

**Reply by robd003** [reply]
@dvmarinoff I opened a ticket with Brave to have them enable Web Bluetooth.

Feel free to upvote here: https://github.com/brave/brave-browser/issues/34941

**Reply by robd003** [reply]
@dvmarinoff Did not see a red banner with "this browser is NOT supported" on Brave.

**Comment by dvmarinoff** [comment]
I am on MacOS Sonoma 14.1.2 with Google Chrome Version 120.0.6099.71 (Official Build) (arm64).

**Reply by robd003** [reply]
Works fine on Chrome, it's a Brave (_Chromium based browser that has really good ad blocking written in Rust_) issue.

I guess I'll leave this around so other Brave users don't wonder why it's not working.

**Reply by dvmarinoff** [reply]
Thanks that's pretty important issue, I have multiple other web-bluetooth projects and didn't know about it, will notify everyone.
